autowrap/PXDParser.py

```
@@ -398,8 +412,13 @@ def parseTree(cls, node: Cython.Compiler.Nodes.CppClassNode, lines: Collection[s
             # template argument is required or optional.
             # For now, convert to pre-0.24 format
             template_parameters = [t[0] for t in template_parameters]
-
-        class_annotations = parse_class_annotations(node, lines)
+        try:
+            class_annotations = parse_class_annotations(node, lines)
+        except ValueError as e:
+            logger.warning(
+                " Failed to parse class annotations in %s:%d\n%s", pxd_path, node.pos[1], e
+            )
+            raise ValueError(f"Failed to parse class annotations in {pxd_path}:{node.pos[1]}\n{e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)